emcc.py

```diff
@@ -1151,11 +1151,14 @@ def check(input_file):
 
     if shared.Settings.MAIN_MODULE:
       assert not shared.Settings.SIDE_MODULE
-      if shared.Settings.MAIN_MODULE != 2:
+      if shared.Settings.MAIN_MODULE == 1:
         shared.Settings.INCLUDE_FULL_LIBRARY = 1
+        shared.Settings.EXPORT_ALL = 1
     elif shared.Settings.SIDE_MODULE:
       assert not shared.Settings.MAIN_MODULE
       options.memory_init_file = False # memory init file is not supported with asm.js side modules, must be executable synchronously (for dlopen)
+      if not shared.Settings.WASM_BACKEND:
+        shared.Settings.EXPORT_ALL = 1
```

Comment:
I think you can just do this once down on line 1166, no?   Right below where we set `LINKABLE = 1`?

Reply:
Still needs doing.

Reply:
Thanks, done.